elastic/kibana
# [Security Solution][Detections] - Fix threshold preview

## Summary

Addresses https://github.com/elastic/kibana/issues/92732

7.11+ versions of threshold preview histogram were aggregating by "event.category". This PR updates the preview histogram to take into account threshold field groups and cardinality.

It may need to be called out in documentation or updated to remind users that preview is not an exact guarantee of what signals will be produced as it does not take into account `interval` and any `timestamp_override`. Threshold gets a tad bit more confusing because of the multiple aggregations occurring (threshold --> group by field --> histogram).

#### To test
 - Go to create a threshold rule
 - Attempt various combos such as:
   -  Threshold rule with query and threshold value only
       -  What is being asked: generate an alert for every time within an interval query matches with `min_doc_count` equal to or greater than threshold
   - Threshold rule with query, threshold field and threshold value
     -  What is being asked: generate an alert for every specified threshold field (for example `source.ip`) that appears in at least 'x' search results within time interval
   - Threshold rule with query, multiple threshold fields and threshold value
     - What is being asked: generate an alert [FIELD_1] **AND** [FIELD_2] appear in at least 'x' search results within time interval
   - Threshold rule with query, threshold field, threshold value, and cardinality values filled out
     -  What is being asked: generate an alert for every specified threshold field (for example `source.ip`) that appears in at least 'x' search results AND where the specified field for cardinality has the specified number of distinct values 

### Checklist
